core/src/main/java/com/linecorp/armeria/internal/common/RequestContextUtil.java

```diff
@@ -195,6 +202,11 @@ public static SafeCloseable invokeHookAndPop(RequestContext current, @Nullable R
         }
     }
 
+    public static boolean equalUnwrapAllNullable(@Nullable RequestContext ctx1, @Nullable RequestContext ctx2) {
+        return Optional.ofNullable(ctx1).map(RequestContext::unwrapAll).orElse(null) ==
+               Optional.ofNullable(ctx2).map(RequestContext::unwrapAll).orElse(null);
+    }
```

Comment:
This will create additional instances every time the method is called. How about
```suggestion
    public static boolean equalsUnwrapping(@Nullable RequestContext ctx1, @Nullable RequestContext ctx2) {
        if (ctx1 == ctx2) {
            return true;
        }
        if (ctx1 != null && ctx2 != null) {
            return ctx1.unwrapAll() == ctx2.unwrapAll();
        }
        return false;
    }
```

Reply:
I can hardly say that it is convenient/easy to compare wrapped objects with this utility method.
How about adding a helper method to `Unwrappable` like
```java
interface Unwrappable {
    // The method name was inspired by String.equalsIgnoreCase().
    // Better naming is welcome. 😆
    @UnstableApi
    boolean equalsIgnoreWrapper(@Nullable Unwrappable other) {

    }
}
```
so that we can easily compare the `ctx`s without complex if else conditions that might distract other important code.